app/src/main/java/eu/kanade/tachiyomi/util/chapter/ChapterSort.kt

```diff
@@ -14,14 +14,18 @@ class ChapterSort(val manga: Manga, val chapterFilter: ChapterFilter = Injekt.ge
         filterForReader: Boolean = false,
         currentChapter: T? = null
     ): List<T> {
+        val filteredChapterList = manga.filtered_scanlators?.let { filteredScanlatorString ->
+            val filteredScanlators = ChapterUtil.getScanlators(filteredScanlatorString)
+            return rawChapters.filter { ChapterUtil.getScanlators(it.scanlator).any { group -> filteredScanlators.contains(group) } }
+        } ?: rawChapters
```

Comment:
just noticed, this also need to be added to 1getNextUnreadChapter` method below as well

Reply:
I actually couldn't get this to work for some reason. Doing a copy and paste gives an error because it wanted a T? and when I tried to fix it, it would never actually work when I tested :(

Reply:
I'm assuming [this](https://github.com/Jays2Kings/tachiyomiJ2K/blob/fbcbe879a5245033dfc8b7ba0914d8eb6a85758e/app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderPresenter.kt#L122) is where the chapters are loaded in the reader but it uses the `ChapterFilter.kt` functions so it doesn't hit `ChapterSort.kt` functions for determining to read next but does filter the chapter list to show.